Plugin page update (#563)
PennyLaneAI/pennylane-qiskit: doc/devices/remote.rst

The ``'qiskit.remote'`` device is a generic adapter to use any Qiskit backend as interface
for a PennyLane device.

To access IBM backends, we recommend using `Qiskit Runtime <https://docs.quantum.ibm.com/api/migration-guides/qiskit-runtime-from-ibm-provider>`_.

.. code-block:: python

    from qiskit_ibm_runtime import QiskitRuntimeService

>     QiskitRuntimeService.save_account(channel="ibm_quantum", token="<IQP_TOKEN>")

Reviewer: Why do we call it `<IQP_>`? What is IQP an acronym for?

Author: IBM Quantum Platform (?) I think. It's copy and pasted from Qiskit's own documentation, but I could replace it with a more informative name like "your_ibm_q_token"

Reviewer: It should be okay, I was just curious lol.

    # To access saved credentials for the IBM quantum channel and select an instance
    service = QiskitRuntimeService(channel="ibm_quantum", instance="my_hub/my_group/my_project")
    backend = service.least_busy(operational=True, simulator=False, min_num_qubits=<num_qubits>)

    dev = qml.device('qiskit.remote', wires=<num_qubits>, backend=backend)

.. note:: 

    Certain third-party backends may be using the deprecated ``Provider`` interface, in which case
    you can get the backend instance from providers with complex search options using their 
    ``get_backend()`` method. For example:

    .. code-block:: python

        import pennylane as qml

        def configured_backend():
            backend = SomeProvider.get_backend(...)
        backend.options.update_options(...) # Set backend options this way
        return backend

        dev = qml.device('qiskit.remote', wires=2, backend=configured_backend())

After installing the plugin, this device can be used just like any other PennyLane device for defining and evaluating QNodes.
For example, a simple quantum function that returns the expectation value of a measurement and depends on

> three classical input parameters can be decorated with ``qml.qnode`` as usual  to construct a ``QNode``:

Reviewer: ```suggestion
three classical input parameters can be decorated with ``qml.qnode`` as usual to construct a ``QNode``:
```

.. code-block:: python

    @qml.qnode(dev)
    def circuit(x, y, z):
        qml.RZ(z, wires=[0])
        qml.RY(y, wires=[0])
        qml.RX(x, wires=[0])
        qml.CNOT(wires=[0, 1])
        return qml.expval(qml.PauliZ(wires=1))

> You can then execute the above quantum circuit to get the quantum mechanical expectation value.

Reviewer: ```suggestion
You can then execute the above quantum circuit to get the expectation value of a Pauli operator.
```

The ``'qiskit.remote'`` device is a generic adapter to use any Qiskit backend as interface
for a PennyLane device.

To access IBM backends, we recommend using `Qiskit Runtime <https://docs.quantum.ibm.com/api/migration-guides/qiskit-runtime-from-ibm-provider>`_.

.. code-block:: python

    from qiskit_ibm_runtime import QiskitRuntimeService

    QiskitRuntimeService.save_account(
        channel="ibm_quantum", 
        token="<IQP_TOKEN>", 
        overwrite=True, 

>         default=true

Reviewer 2: At the least `True` should be capitalized but in my notebook it rejects the `default` kwarg entirely - maybe I have the wrong version? I get `TypeError: QiskitRuntimeService.save_account() got an unexpected keyword argument 'default'`

Author: That... is strange considering I copy pasted this straight off of [Qiskit's documentation](https://docs.quantum.ibm.com/api/migration-guides/qiskit-runtime-from-ibmq-provider) (Not the first time they had a non-working code example). Though... I'm relatively sure I also ran this code myself and it was fine. Let me double check and I'll get back to you.

Author: This is probably the safest way to save an account
`QiskitRuntimeService.save_account(channel="ibm_quantum", token="MY_IBM_QUANTUM_TOKEN")`

Seems like the default keyword is not a keyword that actually exists. Well, just don't read that migration guide I guess :)

    )

    # To access saved credentials for the IBM quantum channel and select an instance
    service = QiskitRuntimeService(channel="ibm_quantum", instance="my_hub/my_group/my_project")
    backend = service.least_busy(operational=True, simulator=False, min_num_qubits=<num_qubits>)

    dev = qml.device('qiskit.remote', wires=<num_qubits>, backend=backend)

This device is also useful when accessing backends from providers with complex search options in
their ``get_backend()`` method, or for setting options on a backend prior to wrapping it as

> PennyLane device.

Author: I see. That's a good point. This should be reworded then, maybe in a note as you mentioned later

	return backend

    dev = qml.device('qiskit.remote', wires=2, backend=configured_backend())

.. warning::

>     Retrieving a backend from a provider has been deprecated and may not be supported. 

Reviewer: Could we direct users towards an alternative in the case of expected deprecation?

Author: The alternative is the preferred use with `QiskitRuntimeService` (which is what the docs lead with). 

The reason I leave this is in is because certain other third-party backends still use this provider interface and I want to keep that available, but for IBM backends I think `QiskitRuntimeService` is the way to go. Maybe I should re-iterate that `QiskitRuntimeService` is what should be used.

> After installing the plugin, this device can be used just like other devices for the definition and evaluation of QNodes within PennyLane.

Reviewer: ```suggestion
After installing the plugin, this device can be used just like any other PennyLane device for defining and evaluating QNodes.
```

> A simple quantum function that returns the expectation value of a measurement and depends on three classical input
> parameters would look like:

Reviewer: ```suggestion
For example, a simple quantum function that returns the expectation value of a measurement and depends on
three classical input parameters can be decorated with `qml.qnode` as usual  to construct a ``QNode``:
```

.. code-block:: python

    @qml.qnode(dev)
    def circuit(x, y, z):
        qml.RZ(z, wires=[0])
        qml.RY(y, wires=[0])
        qml.RX(x, wires=[0])
        qml.CNOT(wires=[0, 1])
        return qml.expval(qml.PauliZ(wires=1))

> You can then execute the circuit like any other function to get the quantum mechanical expectation value.

Reviewer: ```suggestion
You can then execute the above quantum circuit to get the quantum mechanical expectation value.
```

.. code-block:: python

    circuit(0.2, 0.1, 0.3)

> The ``'qiskit.remote'`` device also supports the use of `local simulators <https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/fake_provider>`_ such as FakeManila.

Reviewer: ```suggestion
The ``'qiskit.remote'`` device also supports the use of `local simulators <https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/fake_provider>`_ such as ``FakeManila``.
```

.. code-block:: python
    
    from qiskit_ibm_runtime.fake_provider import FakeManilaV2
    backend = FakeManilaV2()

    # You could use an Aer simulator instead by using the following code:
    # from qiskit_aer import AerSimulator
    # backend = AerSimulator()

    dev = qml.device('qiskit.remote', wires=5, backend=backend)

> Device options
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
Device options
~~~~~~~~~~~~~~
```
